public/app/plugins/datasource/mysql/models.gen.ts

```diff
@@ -0,0 +1,19 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// NOTE: This file will be auto generated from models.cue
+// It is currenty hand written but will serve as the target for cuetsy
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+import { DataQuery, DataSourceJsonData } from '@grafana/data';
+
+export const modelVersion = Object.freeze([1, 0]);
+
+export interface MySQLOptions extends DataSourceJsonData {
+  timeInterval: string;
+}
+
+export type ResultFormat = 'time_series' | 'table';
+
+export interface MySQLQuery extends DataQuery {
```

Comment:
i don't...think it'll be possible to do composition this way, where the composed thing is actually extending the parent. That effectively makes `DataQuery` part of the scuemata family rules for each plugin, rather than being its own, independent thing.

Reply:
yup -- kinda why I make this PR... *maybe* we can only support cue for plugins that activly opt-in to moving the query model to a sub-object :thinking:

Reply:
Y'know, that is one area we need to dive into more - ways we need to be able to bend and degrade the overall schema system in order to accommodate incomplete third-party implementations